rfds/116-Handling-Old-Comments/index.adoc

```diff
@@ -0,0 +1,297 @@
+
+The community has raised multiple suggestions around improving comment handling, mostly to reduce noise on pull requests. Some suggestions include:
+
+. Introducing a _"status/summary"_ comment that acts as a consolidated view of all operations on a PR, reducing fragmentation of information and making it easier for reviewers to assess the state of infrastructure changes.
+. Anchoring a single comment that is updated over time, rather than posting multiple new comments for each operation.
+
+TODO Add pictures for the "status/summary" comment.
+
+== Dependencies
+
+This section describes each services/modules that are going to be affected by this feature.
+
+=== GitHub API
+
+Our only supported VCS as of now.
+
+=== GitLab API
+
+TODO We are working on supporting GitLab and should investigate how to accomplish this there as well.
+
+=== PostgreSQL
+
+Currently, we do not store comment data in PostgreSQL. To implement any of the suggested solutions we would have to at least store:
+
+. The Comment ID and its PR number.
+. The status of a comment (e.g., active, superseded, minimized).
+. Track the generation/version of a comment to maintain its modificatio history.
+
+as for now we have the following tables tracking:
+
+* `github_pull_requests`
+
+[source,sql]
+----
+terrateam=> \d github_pull_requests
+                   Table "public.github_pull_requests"
+   Column    |           Type           | Collation | Nullable | Default 
+-------------+--------------------------+-----------+----------+---------
+ base_branch | text                     |           | not null | 
+ base_sha    | text                     |           | not null | 
+ branch      | text                     |           | not null | 
+ pull_number | bigint                   |           | not null | 
+ repository  | bigint                   |           | not null | 
+ sha         | text                     |           | not null | 
+ state       | text                     |           | not null | 
+ merged_sha  | text                     |           |          | 
+ merged_at   | timestamp with time zone |           |          | 
+ title       | text                     |           |          | 
+ username    | text                     |           |          | 
+----
+
+* `github_pull_requests_map`
+
+[source,sql]
+----
+terrateam=> \d github_pull_requests_map
+              Table "public.github_pull_requests_map"
+    Column     |  Type  | Collation | Nullable |      Default      
+---------------+--------+-----------+----------+-------------------
+ repository_id | bigint |           | not null | 
+ pull_number   | bigint |           | not null | 
+ core_id       | uuid   |           | not null | gen_random_uuid()
+----
+
+=== Terrateam Configuration
+
+. Given that our aim to make the most configurable IaC tool ever, we need to embed a new set of configurations and default values.
+. Focus on making it per-directory so people can have freedom to apply different commenting rules.
```

Comment:
I think this might be hard to do, but I encourage trying some ideas to see if this behaviour can make sense on a per-directory bases.